### PR TITLE
Adding Skriware 2 definitions

### PR DESCRIPTION
**Pull request by skriDude**

### CODE REVIEW

File: resources/extruders/skriware_2_extruder_0.def.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "id": "skriware_2_extruder_0",
```
Comment (Ghostkeeper) [inline resources/extruders/skriware_2_extruder_0.def.json]: These IDs are superfluous. They aren't being used. Could you remove them?

Comment (skriDude) [reply resources/extruders/skriware_2_extruder_0.def.json]: Are the files good to go or still need more changes?

**Comment by Ghostkeeper** [comment]
Looks fine at first glance. We'll test it.

Developers, see issue CURA-6949.

**Comment by Ghostkeeper** [comment]
There are a lot of settings in this definition that don't have any effect since they override a `default_value` for a setting that has a `value` set. The `value` always takes precedence over the `default_value`, so this `default_value` has no effect unless you were running CuraEngine from the command line.

List of setting overrides that don't have any effect: 'acceleration_infill', 'acceleration_ironing', 'acceleration_layer_0', 'acceleration_prime_tower', 'acceleration_print_layer_0', 'acceleration_roofing', 'acceleration_skirt_brim', 'acceleration_support', 'acceleration_support_bottom', 'acceleration_support_infill', 'acceleration_support_interface', 'acceleration_support_roof', 'acceleration_topbottom', 'acceleration_travel', 'acceleration_travel_layer_0', 'acceleration_wall', 'acceleration_wall_0', 'acceleration_wall_x', 'bottom_layers', 'bottom_skin_expand_distance', 'bottom_skin_preshrink', 'bridge_skin_speed_2', 'cool_fan_full_at_height', 'cool_fan_full_layer', 'cross_infill_pocket_size', 'expand_skins_expand_distance', 'extruders_enabled_count', 'gantry_height', 'infill_line_distance', 'infill_material_flow', 'infill_sparse_thickness', 'infill_wipe_dist', 'ironing_inset', 'jerk_infill', 'jerk_ironing', 'jerk_layer_0', 'jerk_prime_tower', 'jerk_print_layer_0', 'jerk_roofing', 'jerk_skirt_brim', 'jerk_support', 'jerk_support_bottom', 'jerk_support_infill', 'jerk_support_interface', 'jerk_support_roof', 'jerk_topbottom', 'jerk_travel', 'jerk_travel_layer_0', 'jerk_wall', 'jerk_wall_0', 'jerk_wall_x', 'layer_0_z_overlap', 'material_bed_temperature', 'material_bed_temperature_layer_0', 'material_initial_print_temperature', 'material_print_temperature', 'material_print_temperature_layer_0', 'meshfix_maximum_travel_resolution', 'min_skin_width_for_expansion', 'prime_tower_flow', 'prime_tower_position_x', 'prime_tower_position_y', 'raft_acceleration', 'raft_base_acceleration', 'raft_base_jerk', 'raft_base_line_spacing', 'raft_base_line_width', 'raft_base_speed', 'raft_interface_acceleration', 'raft_interface_jerk', 'raft_interface_line_spacing', 'raft_interface_line_width', 'raft_interface_speed', 'raft_interface_thickness', 'raft_jerk', 'raft_speed', 'raft_surface_acceleration', 'raft_surface_jerk', 'raft_surface_speed', 'raft_surface_thickness', 'retraction_extrusion_window', 'retraction_min_travel', 'retraction_prime_speed', 'retraction_retract_speed', 'roofing_layer_count', 'roofing_material_flow', 'skin_material_flow', 'skin_overlap', 'skin_overlap_mm', 'skin_preshrink', 'skirt_brim_line_width', 'skirt_brim_material_flow', 'speed_roofing', 'speed_travel', 'speed_travel_layer_0', 'support_bottom_distance', 'support_bottom_material_flow', 'support_brim_line_count', 'support_infill_rate', 'support_infill_sparse_thickness', 'support_initial_layer_line_distance', 'support_interface_material_flow', 'support_line_distance', 'support_material_flow', 'support_roof_density', 'support_roof_height', 'support_roof_line_distance', 'support_roof_material_flow', 'support_top_distance', 'support_tree_collision_resolution', 'support_tree_wall_thickness', 'support_xy_distance_overhang', 'support_zag_skip_count', 'switch_extruder_prime_speed', 'switch_extruder_retraction_speed', 'top_bottom_pattern_0', 'top_layers', 'top_skin_expand_distance', 'top_skin_preshrink', 'wall_0_material_flow', 'wall_line_count', 'wall_material_flow', 'wall_x_material_flow', 'wipe_retraction_prime_speed', 'wipe_retraction_retract_speed', 'xy_offset_layer_0', 'z_seam_x', 'z_seam_y'

So the question is then: Did you actually test the definition file you are submitting or did you test the setting values?

If you tested the definition file, you should remove these setting overrides that don't have any effect. If you tested the setting values, you need to change the `default_value` overrides into `value` overrides.

**Comment by skriDude** [comment]
Thanks for explaining and pointing it out! 

What we did is we tested the settings in CuraGUI and then with a custom plugin exported those settings into a definition file with all parameters as `default_value`.

I'm updating the file with the parameters you've mentioned changed from `default_value` to `value`.

**Comment by fieldOfView** [comment]
This won't work as-is either.
The "value" must always be a string, which is interpreted by Python. So instead of `5`, the values should be `"5"`.

**Comment by Ghostkeeper** [comment]
I believe we have a fall-back that interprets them as strings, so it would've worked. At least, I've seen it quite often in other definitions. But strings is more appropriate.

**Comment by skriDude** [comment]
Changed to strings already :)

**Comment by Ghostkeeper** [comment]
It's just complaining about `acceleration_ironing` now. I'll change that for you.

It's also logging some warnings when loading this definition file or when running the automated tests:

```
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting machine_extruder_start_pos_x
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting machine_extruder_cooling_fan_number
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting machine_extruder_start_pos_y
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting machine_extruder_start_code
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting machine_nozzle_offset_y
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting machine_extruder_end_pos_y
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting machine_extruder_end_pos_x
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting extruder_nr
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting machine_extruder_end_code
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting machine_extruder_start_pos_abs
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting machine_nozzle_offset_x
1: [MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [270]: Unable to override setting machine_extruder_end_pos_abs
```

Those are about settings that are exclusive to the extruder. These all happen to be equal to the defaults so I'll remove them.

**Comment by Ghostkeeper** [comment]
See my changes here: https://github.com/Ultimaker/Cura/commit/74d52bbd1ca744cbb5fbbf82f84cac2c6383f8c9

**Comment by skriDude** [comment]
Thanks for all the help!

**Comment by skriDude** [comment]
Hey @Ghostkeeper, I'd like to ask why Skriware 2 doesn't appear in Cura 4.4? Were there some problems with the definitions?

**Comment by Ghostkeeper** [comment]
No, it's just that we were too late with reviewing it to get it done in time for Cura 4.4. Version 4.4 went into feature freeze on Halloween iirc. so it was merged into master to be released with Cura 4.5 instead. Sorry.

**Comment by skriDude** [comment]
Alright, thanks for the update!

**Pull request merged**